lib/spack/spack/spec.py

```diff
@@ -1560,6 +1567,19 @@ def from_node_dict(node):
             spec.external_path = None
             spec.external_module = None
 
+        if 'concrete' in node:
```

Comment:
@citibeth if this is addressing https://github.com/spack/spack/pull/4585#issuecomment-382903792, then this looks like it may be a more general bug and the fix ought to go in its own PR. Would you mind doing that?